cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/PrecompileCacheTest.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * (c) Copyright 2016 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package io.cloudslang.lang.compiler;
+
+import io.cloudslang.lang.compiler.caching.CachedPrecompileService;
+import io.cloudslang.lang.compiler.modeller.SlangModeller;
+import io.cloudslang.lang.compiler.parser.YamlParser;
+import io.cloudslang.lang.compiler.scorecompiler.ScoreCompiler;
+import io.cloudslang.lang.compiler.validator.CompileValidator;
+import io.cloudslang.lang.compiler.validator.SystemPropertyValidator;
+import java.net.URL;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.mockito.Mockito.inOrder;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PrecompileCacheTest {
```

Comment:
agree, however `SlangCompilerImplTest` is already taken - it's a mix between mocks and real objects - the merge is not straightforward - I think I'll leave it like this for now